File: src/main/java/intellistart/interviewplanning/model/candidateslot/CandidateSlot.java
package intellistart.interviewplanning.model.candidateslot;

import intellistart.interviewplanning.model.booking.Booking;
import intellistart.interviewplanning.model.period.Period;
import java.time.LocalDate;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

/**
 * CandidateSlot entity.
 */
@Entity
@Table(name = "candidate_slots")
@NoArgsConstructor
@AllArgsConstructor
public class CandidateSlot {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "candidate_slot_id")
  private Long id;

  private LocalDate date;

  @ManyToOne
  @JoinColumn(name = "period_id")
  private Period period;

  @OneToMany(mappedBy = "candidateSlot")
  private Set<Booking> bookings = new HashSet<>();

  private String email;
  private String name;

  public void addBooking(Booking booking) {
    bookings.add(booking);
  }

  @Override
  public String toString() {
    return "CandidateSlot{"
        + "id=" + id
        + ", date=" + date
        + ", period=" + period
        + ", email='" + email
        + '\'' + ", name='"
        + name + '\'' + '}';
  }

  public Long getId() {
    return id;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public LocalDate getDate() {
    return date;
  }

  public void setDate(LocalDate date) {
    this.date = date;
  }

  public Period getPeriod() {
    return period;
  }

  public void setPeriod(Period period) {
    this.period = period;
  }

  public Set<Booking> getBookings() {
    return bookings;
  }

  public void setBookings(Set<Booking> bookings) {
    this.bookings = bookings;
  }

  public String getEmail() {
    return email;
  }

  public void setEmail(String email) {
    this.email = email;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    CandidateSlot that = (CandidateSlot) o;
    return Objects.equals(id, that.id);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id);
  }
}

File: src/main/java/intellistart/interviewplanning/model/candidateslot/CandidateSlot.kt
package intellistart.interviewplanning.model.candidateslot

import intellistart.interviewplanning.model.booking.Booking
import intellistart.interviewplanning.model.period.Period
import java.time.LocalDate
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.persistence.Table

/**
 * CandidateSlot entity.
 */
@Entity
@Table(name = "candidate_slots")
data class CandidateSlot(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "candidate_slot_id")
    var id: Long = 0,

    var date: LocalDate = LocalDate.now(),

    @ManyToOne
    @JoinColumn(name = "period_id")
    var period: Period = Period(),

    @OneToMany(mappedBy = "candidateSlot")
    var bookings: MutableSet<Booking> = HashSet(),

    var email: String = "",

    var name: String = ""
) {
    override fun hashCode(): Int {
        return id.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as CandidateSlot

        if (id != other.id) return false
        if (date != other.date) return false
        if (period != other.period) return false
        if (bookings != other.bookings) return false
        if (email != other.email) return false
        if (name != other.name) return false

        return true
    }
}
